### PR TITLE
Upgrade from Chromium 108 to Chromium 109.

### PR DESCRIPTION
**Pull request by mkarolin**
Resolves https://github.com/brave/brave-browser/issues/26376

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: vendor/bat-native-ads/src/bat/ads/internal/ads/serving/eligible_ads/exclusion_rules/split_test_exclusion_rule_unittest.cc
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/ads/serving/eligible_ads/exclusion_rules/split_test_exclusion_rule_unittest.cc:24]: Rather than `0.9` should we use `RandDouble()` to match the previous provider?

**Review by tmancey — APPROVED** [review]
LGTM++

File: vendor/bat-native-ads/src/bat/ads/internal/ads/serving/eligible_ads/exclusion_rules/split_test_exclusion_rule_unittest.cc
Comment (mkarolin) [reply vendor/bat-native-ads/src/bat/ads/internal/ads/serving/eligible_ads/exclusion_rules/split_test_exclusion_rule_unittest.cc:24]: From what I can tell we just care that a FieldTrial exists. Upstream seems to use specific values in their tests, but I am not sure if it would make a difference for these tests. WDYT, @cdesouza-chromium?

File: chromium_src/components/sync/base/model_type_unittest.cc
Comment (AlexeyBarabash) [inline chromium_src/components/sync/base/model_type_unittest.cc:17]: new types are
```
  // Segmentation data.
  SEGMENTATION,

  // A tab group saved by a user. Currently only supported on desktop platforms
  // (Linux, Mac, Windows, ChromeOS).
  SAVED_TAB_GROUP,
```
`SEGMENTATION` if for Google Analytics, it is not enabled for us even if we set `Sync everything` 
`SAVED_TAB_GROUP` appears to be synced through `Open tabs`, it does not require the additional switch , but it doesn't work for us.

Follow-up  issues https://github.com/brave/brave-browser/issues/27240, https://github.com/brave/brave-browser/issues/27241

**Review by AlexeyBarabash — APPROVED** [review]
Sync changes look good, brief test of Sync Desktop<=>Android also passed.

**Review by yrliou — APPROVED** [review]
wallet++

File: third_party/rust/cxx/v1/crate/Cargo.toml.orig
Comment (antonok-edm) [inline third_party/rust/cxx/v1/crate/Cargo.toml.orig:2]: Is it possible to handle this as a submodule or `DEPS` entry rather than embedding the entire source tree directly into our git repo?

File: chromium_src/chrome/browser/ui/webui/favicon_source.cc
Comment (goodov) [inline chromium_src/chrome/browser/ui/webui/favicon_source.cc:29]: I believe this should be:
```
namespace extensions {
namespace {
```

otherwise these classes may conflict with similar ones defined somewhere else.

File: third_party/rust/cxx/v1/crate/.bazelignore
Comment (goodov) [inline third_party/rust/cxx/v1/crate/.bazelignore:1]: why are we putting this into our repository instead of using `DEPS` like so https://github.com/brave/brave-core/blob/f79211ec4cb1786ddf8ee9793d22b4166c929765/DEPS#L25?

This whole dir looks like a single git dependency:
https://github.com/brave/brave-core/tree/cr109/third_party/rust/cxx/v1/crate

File: patches/ios-chrome-browser-application_context-BUILD.gn.patch
Comment (bridiver) [inline patches/ios-chrome-browser-application_context-BUILD.gn.patch]: we shouldn't be including brave/ios/browser in ios/chrome/browser/application_context, we need to move things out of ios/chrome/browser to match what upstream has done

File: chromium_src/chrome/browser/ui/webui/favicon_source.cc
Comment (mkarolin) [reply chromium_src/chrome/browser/ui/webui/favicon_source.cc:29]: Updated in fec11b38ff

File: patches/extensions-common-api-_permission_features.json.patch
Comment (bridiver) [inline patches/extensions-common-api-_permission_features.json.patch:9]: fyi - https://github.com/brave/brave-core/pull/9061/files#r1046314436

File: patches/ios-chrome-browser-application_context-BUILD.gn.patch
Comment (bridiver) [inline patches/ios-chrome-browser-application_context-BUILD.gn.patch:9]: see previous comment, this is going to create a mess with dependencies

File: patches/ios-chrome-browser-application_context-BUILD.gn.patch
Comment (mkarolin) [reply patches/ios-chrome-browser-application_context-BUILD.gn.patch]: It looks like all this dep is doing is adding a dep on `//brave/common` and the only place I see it's needed in for iOS there is in the `brave/chromium_src/ios/chrome/browser/net/ios_chrome_network_delegate.cc` override. So maybe we can just add a `sources.gni` to `brave/chromium_src/ios/chrome/browser/net` and patch the corresponding upstream BUILD.gn?

File: patches/google_apis-google_api_keys.cc.patch
Comment (bridiver) [inline patches/google_apis-google_api_keys.cc.patch:9]: I don't think this patch is necessary and it seems to need manual updating frequently based on history. @yrliou the original is here https://github.com/brave/brave-core/commit/70def0c486d9945c7be1946b346229156492be9e but I'm not seeing why this is necessary.

File: third_party/rust/cxx/v1/crate/Cargo.toml.orig
Comment (cdesouza-chromium) [reply third_party/rust/cxx/v1/crate/Cargo.toml.orig:2]: This was copied from chromium where it is also checked out inside the repo, with patches applied to it.

File: patches/extensions-common-api-_permission_features.json.patch
Comment (mkarolin) [reply patches/extensions-common-api-_permission_features.json.patch]: Moved to our override in 26abd06a0409fc18aecebe4ae305361d9c59c733. Will create a follow up issue for using chromium_src override.

Comment (mkarolin) [reply patches/extensions-common-api-_permission_features.json.patch]: Follow up issue: https://github.com/brave/brave-browser/issues/27314

File: patches/chrome-browser-upgrade_detector-upgrade_detector_impl.cc.patch
Comment (bridiver) [inline patches/chrome-browser-upgrade_detector-upgrade_detector_impl.cc.patch:13]: switch_name removal seems weird https://github.com/brave/brave-core/commit/1af9646975f07322351a14c75113d5e62ed64919#diff-e1717757fbaa5782d6822820e62e9739aa87b5c766b0560ec00e450766322be3R9

File: patches/chrome-browser-upgrade_detector-upgrade_detector_impl.cc.patch
Comment (bridiver) [inline patches/chrome-browser-upgrade_detector-upgrade_detector_impl.cc.patch:17]: I think maybe we just want to set `GOOGLE_CHROME_BRANDING` to true for this file? Don't we also want `ShouldDetectOutdatedBuilds` to return true?

File: chromium_src/chrome/browser/ui/startup/startup_browser_creator.cc
Comment (bridiver) [inline chromium_src/chrome/browser/ui/startup/startup_browser_creator.cc:28]: how are we going to catch when they add the new launch mode recorder before they removed the old one?

File: patches/ios-chrome-browser-application_context-BUILD.gn.patch
Comment (bridiver) [reply patches/ios-chrome-browser-application_context-BUILD.gn.patch]: we don't want sources.gni unless there's a circular dep so it sounds like we just need to move `//brave/ios/browser:chromium_src` -> `//brave/ios/browser/application_context:chromium_src`

Comment (bridiver) [reply patches/ios-chrome-browser-application_context-BUILD.gn.patch]: wait, why are we adding this to ios/chrome/browser/application then?

File: patches/google_apis-google_api_keys.cc.patch
Comment (yrliou) [reply patches/google_apis-google_api_keys.cc.patch:9]: It's a while back and yea it's very likely there are other ways to do this now, cc @atuchin-m who is now much more familiar with our translate feature for removing this.

File: patches/chrome-browser-upgrade_detector-upgrade_detector_impl.cc.patch
Comment (mkarolin) [reply patches/chrome-browser-upgrade_detector-upgrade_detector_impl.cc.patch]: Yes, this looks to be an accident. Removed this patch in a2aca8d72c95178c93edb8173f5940279a8b3f1d

File: patches/ios-chrome-browser-application_context-BUILD.gn.patch
Comment (mkarolin) [reply patches/ios-chrome-browser-application_context-BUILD.gn.patch]: Fixed via 826b3f59b16e8b259fbfdc52c706d5ab401f62de (+fixups)

File: third_party/rust/cxx/v1/crate/Cargo.toml.orig
Comment (cdesouza-chromium) [reply third_party/rust/cxx/v1/crate/Cargo.toml.orig]: Fixed in 1961e09dad30d0560f2f1e3208643279d2ed3fe2

File: third_party/rust/cxx/v1/crate/.bazelignore
Comment (cdesouza-chromium) [reply third_party/rust/cxx/v1/crate/.bazelignore]: Fixed in 1961e09dad30d0560f2f1e3208643279d2ed3fe2

File: DEPS
Comment (bridiver) [inline DEPS:26]: we can't do this, we need upstream's patches

File: third_party/rust/cxx/v1/crate/.bazelignore
Comment (cdesouza-chromium) [reply third_party/rust/cxx/v1/crate/.bazelignore:1]: The fix above had to be reversed as the version checked out in the tree comes from Chromium, and it has patches applied to it that we need.

File: third_party/rust/cxx/v1/crate/Cargo.toml.orig
Comment (cdesouza-chromium) [reply third_party/rust/cxx/v1/crate/Cargo.toml.orig:2]: The fix above had to be reversed as the version checked out in the tree comes from Chromium, and it has patches applied to it that we need.

File: DEPS
Comment (cdesouza-chromium) [reply DEPS]: Reverted.

File: third_party/rust/cxx/v1/crate/Cargo.toml.orig
Comment (antonok-edm) [reply third_party/rust/cxx/v1/crate/Cargo.toml.orig:2]: I see. It'd be good to document this somewhere at least, so that
a) it's easier to identify how to keep this up to date when the Chromium version changes
b) we can put it in DEPS and remove it from the source tree when Chromium's changes will (presumably) be upstreamed

Comment (cdesouza-chromium) [reply third_party/rust/cxx/v1/crate/Cargo.toml.orig:2]: I would advise doing this separately as a PR. Checking out this dependency in tree is related to an issue linking rust code on mac, which prevents upgrading the toolset to rustc 1.60, so some better understanding of how to move forward has to be reached. This has been added for now to unblock cr109.

File: DEPS
Comment (goodov) [reply DEPS]: can't we just copy these patches and apply ourselves via apply_patches infrastructure?

Comment (cdesouza-chromium) [reply DEPS]: Like I mentioned somewhere else, the cxx sources are also pushed on Chromium, from where I lifted it from. A process to have a DEPS checked out, and patched, although not impossible, would require certain planning, specially as the dependency would seem to have a dirty tree.

I would advise to have this looked into outside the version upgrade PR, as I suspect that we will block the PR further with this rust issue.

File: patches/google_apis-google_api_keys.cc.patch
Comment (atuchin-m) [reply patches/google_apis-google_api_keys.cc.patch:9]: In fact the method is still used in unit_tests: https://github.com/brave/brave-core/blob/master/chromium_src/components/translate/core/browser/translate_manager_unittest.cc#L173

But `google_apis::HasAPIKeyConfigured()` [has a custom override](https://github.com/brave/brave-core/blob/master/chromium_src/components/translate/core/browser/translate_manager.cc#L14) for `TranslateManager`. IMHO we don't need to test set or unset API key in unit_tests, just the main case that it works by default.

So I'm okay to remove it, but the tests need to be changed.

File: chromium_src/chrome/browser/ui/startup/startup_browser_creator.cc
Comment (cdesouza-chromium) [reply chromium_src/chrome/browser/ui/startup/startup_browser_creator.cc:28]: We are going to have a build breakage in the call of `StartupBrowserCreatorImpl::Launch`, whenever the new one gets rolled out into `StartupBrowserCreatorImpl`.

Comment (bridiver) [reply chromium_src/chrome/browser/ui/startup/startup_browser_creator.cc:28]: maybe something like `#ifdef LaunchModeRecorder static_assert(false) #endif`?

Comment (cdesouza-chromium) [reply chromium_src/chrome/browser/ui/startup/startup_browser_creator.cc:28]: That might or might not matter. The key here is whatever type is being used as an argument in the signature for `StartupBrowserCreatorImpl::Launch`. I'm gonna add it though as it won't hurt, but it may be the case that at some point we have a new `LaunchModeRecorder` and yet an outdated `StartupBrowserCreatorImpl::Launch` signature.

Comment (cdesouza-chromium) [reply chromium_src/chrome/browser/ui/startup/startup_browser_creator.cc:34]: Fixed in c715672b5584f46a9b6325748f5fa52383890722

File: patches/google_apis-google_api_keys.cc.patch
Comment (mkarolin) [reply patches/google_apis-google_api_keys.cc.patch:9]: Create a follow up issue https://github.com/brave/brave-browser/issues/27338

**Review by fmarier — APPROVED** [review]
`script/brave_license_helper.py` looks good

**Review by antonok-edm — APPROVED** [review]

**Review by petemill — APPROVED** [review]
Looks like WebUI changes were minimal and done by me 😉

**Comment by mkarolin** [comment]
**When uplifting, also uplift:**
* https://github.com/brave/brave-browser/issues/27424

**Pull request merged**
